meta/Meta.cpp

```diff
@@ -7006,43 +7006,43 @@ void Meta::meta_sai_on_fdb_flush_event_consolidated(
     }
 }
 
-void Meta::meta_fdb_event_snoop_oid(
+int Meta::meta_fdb_event_snoop_oid(
         _In_ sai_object_id_t oid)
 {
     SWSS_LOG_ENTER();
 
     if (oid == SAI_NULL_OBJECT_ID)
-        return;
+        return -1;
 
     if (m_oids.objectReferenceExists(oid))
-        return;
-
-    sai_object_type_t ot = objectTypeQuery(oid);
-
-    if (ot == SAI_OBJECT_TYPE_NULL)
-    {
-        SWSS_LOG_ERROR("failed to get object type on fdb_event oid: 0x%" PRIx64 "", oid);
-        return;
-    }
-
-    sai_object_meta_key_t key = { .objecttype = ot, .objectkey = { .key = { .object_id = oid } } };
-
-    m_oids.objectReferenceInsert(oid);
-
-    if (!m_saiObjectCollection.objectExists(key))
-        m_saiObjectCollection.createObject(key);
+        return 0;
 
     /*
      * In normal operation orch agent should query or create all bridge, vlan
      * and bridge port, so we should not get this message. Let's put it as
      * warning for better visibility. Most likely if this happen  there is a
      * vendor bug in SAI and we should also see warnings or errors reported
      * from syncd in logs.
+     *
+     * Originally the solution was to create the object and then continue to
+     * learn fdb_entry, which causes other issue on SONiC. Consider the following
+     * scenario:
+     * 1. SWITCH ASIC reports fdb learnt on a certain vlan/port
+     * 2. Create object for the vlan/port
+     * 3. Learn the mac address, which will increase the reference count of port
+     * 4. On SONiC, all port will be removed from bridge during creating switch,
```

Comment:
learning fdb entry during switch create i perfectly normal scenario, so learn should happen even during switch create. If some fdb entiies were learned during that time, and reference was increased, then is up to user to remove those fdb entries before removing port

Reply:
Do you mean this issue should be handled in orchagent where the port is removed?

Reply:
yes, that is my intention